src/DeviceInterface/ExtensionInterface.js

```diff
@@ -27,31 +29,64 @@ class ExtensionInterface extends InterfacePrototype {
             testMode: this.isTestMode()
         }
         const tooltipKinds = {
-            [POPUP_TYPES.EmailProtection]: 'legacy',
-            [POPUP_TYPES.EmailSignup]: 'emailsignup'
+            [TOOLTIP_TYPES.EmailProtection]: 'legacy',
+            [TOOLTIP_TYPES.EmailSignup]: 'emailsignup'
         }
-        const tooltipKind = tooltipKinds[this.getShowingTooltip()] || tooltipKinds[POPUP_TYPES.EmailProtection]
+        const tooltipKind = tooltipKinds[this.getShowingTooltip()] || tooltipKinds[TOOLTIP_TYPES.EmailProtection]
 
         return new HTMLTooltipUIController({ tooltipKind, device: this }, htmlTooltipOptions)
     }
 
-    get hasDismissedEmailSignup () {
-        // TODO -- implement peristed dismissed timestamp
-        return false
-    }
-
     getShowingTooltip () {
```

Comment:
From reading it sounds like it may return the actual tooltip instance, rather than the type. How could we rename it? This is used to understand what tooltip we will show, so it could be something like `determineTooltipType` or even `determineTooltipTypeToShow`? WDYT?